include/server/mir/scene/surface.h

```diff
@@ -142,6 +142,16 @@ class Surface :
 
     /// The session this surface was created by
     virtual auto session() const -> std::weak_ptr<Session> = 0;
+
+    /// Sets the dimensions of the window frame
+    /// It is assumed that any decorations are provided by a child window
+    /// Frame size is not directly accessable, but can be determined by comparing client_size() and frame_size()
+    /// This will trigger a resize in which the client size is adjusted and the frame size remains unchanged
+    virtual void set_frame_geometry(
+        geometry::Height top,
+        geometry::Width left,
+        geometry::Height bottom,
+        geometry::Width right) = 0;
```

Comment:
Unless I misunderstand the intent, these should be `DeltaY` & `DeltaX`, not `Width` & `Height`?